### PR TITLE
docs: OinkOink example game

### PR DESCRIPTION
**Pull request by helios1138**

### CODE REVIEW

File: examples/oink-oink/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Sudoku
```
Comment (bfelbo) [inline examples/oink-oink/README.md]:
```suggestion
# Oink Oink
```

File: examples/oink-oink/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Think you know what a sad frog sounds like? Test your impressions against your friends!
+Here's a [demo](https://developers.rune.ai/examples/oink-oink/).
+
+[<img src="../../docs/static/img/multiplayer-games/oink-oink.png" width=500>](https://developers.rune.ai/examples/sudoku/)
```
Comment (bfelbo) [inline examples/oink-oink/README.md]:
```suggestion
[<img src="../../docs/static/img/multiplayer-games/oink-oink.png" width=500>](https://developers.rune.ai/examples/oink-oink/)
```

File: examples/oink-oink/src/logic.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+export const numRounds = 3
+export const turnCountdown = 3
+
+export const turnDuration = 30
+
+export const turnAlmostOverAt = 5
+
+export const endOfTurnDuration = 3
+
+export const displayCorrectGuessFor = 3
```
Comment (bfelbo) [inline examples/oink-oink/src/logic.ts]:
```suggestion
export const turnDuration = 30
export const turnAlmostOverAt = 5
export const endOfTurnDuration = 3
export const displayCorrectGuessFor = 3
```

FFTI

File: examples/oink-oink/src/logic.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            ),
+            delayPopUp: true,
+          })
+        }
+      } else {
+        setActor(game, "next")
+        newTurn(game)
+      }
+    }
+  },
```
Comment (bfelbo) [inline examples/oink-oink/src/logic.ts]:
```suggestion
  update: ({ game }) => {
    if (!game.currentTurn || !game.currentTurn.timerStartedAt) return

    switch (game.currentTurn.stage) {
      case "countdown":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + turnCountdown
        ) {
          game.currentTurn.stage = "acting"
          game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
        }
        return
      case "acting":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + turnDuration
        ) {
          game.currentTurn.stage = "endOfTurn"
          game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
        }
        return
      case "endOfTurn":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + endOfTurnDuration
        ) {
          for (const player of game.players) {
            player.latestTurnScore = 0
          }

          if (isLastActor(game)) {
            game.currentTurn.stage = "result"

            if (game.round + 1 === numRounds) {
              game.gameOver = true
              Rune.gameOver({
                players: game.players.reduce(
                  (acc, player) => ({
                    ...acc,
                    [player.id]: player.score,
                  }),
                  {}
                ),
                delayPopUp: true,
              })
            }
          } else {
            setActor(game, "next")
            newTurn(game)
          }
        }
        return
    }
  },
```

IMO we can refactor this to make it clearer by:
- moving up `!game.currentTurn.timerStartedAt` into an early return
- converting the if conditions to a switch case

I think this helps make it clear that the logic in the stages are mutually exclusive and see what the logic is for each stage to move to the next. But it's a matter of personal preference 😄

**Review by bfelbo — APPROVED** [review]
Awesome to add this party game example 🥳

File: examples/oink-oink/src/logic.ts (hunk 4 above)
Comment (helios1138) [reply examples/oink-oink/src/logic.ts]: Good idea, thanks! Done

**Pull request merged**